### PR TITLE
Get signature and signature algorithm from CoseSign1_Object

### PR DESCRIPTION
**Pull request by peterssond**

### CODE REVIEW

File: create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java [hunk 1]
```diff
@@ -350,11 +373,10 @@ public void verifySignature(final PublicKey publicKey) throws SignatureException
 
     // First find out which algorithm to use by searching for the algorithm ID in the protected attributes.
     //
-    final CBORObject registeredAlgorithm = this.protectedAttributes.get(HeaderParameterKey.ALG.getCborObject());
-    if (registeredAlgorithm == null) {
+    final SignatureAlgorithm algorithm = getSignatureAlgorithm();
```
Comment (martin-lindstrom) [inline create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java:376]: this.getSignatureAlgorithm()

**Review by martin-lindstrom — APPROVED** [review]
Some cosmetic comments.

File: create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java [hunk 2]
```diff
@@ -350,11 +373,10 @@ public void verifySignature(final PublicKey publicKey) throws SignatureException
...
+    if(algorithm == null) {
```
[the remark this answers is not in the document]
Comment (peterssond) [reply create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java]: Fixed

File: create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java (hunk 1 above)
Comment (peterssond) [reply create-validate/src/main/java/se/digg/dgc/signatures/cose/CoseSign1_Object.java]: Fixed

**Pull request merged**